Fix api documentation for System.Runtime.Intrinsics

dakersnar opened the pull request:
Fixes https://github.com/dotnet/runtime/issues/72402

gewarren commented on xml/System.Runtime.Intrinsics/Vector128`1.xml:
```diff
@@ -37,8 +37,8 @@
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
-    <summary>To be added.</summary>
+    <typeparam name="T">The vector type. <c>T</c> can be any primitive numeric type.</typeparam>
```
```suggestion
    <typeparam name="T">The vector type. <typeparamref name="T"> can be any primitive numeric type.</typeparam>
```

gewarren commented on xml/System.Runtime.Intrinsics/Vector256`1.xml:
```diff
@@ -37,8 +37,8 @@
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
-    <summary>To be added.</summary>
+    <typeparam name="T">The vector type. <c>T</c> can be any primitive numeric type.</typeparam>
```
```suggestion
    <typeparam name="T">The vector type. <typeparam name="T"> can be any primitive numeric type.</typeparam>
```

dakersnar replied:
I assume you mean "typeparamref" here too, right? And below.

dakersnar replied:
Also, these should have a slash at the end, right? Like `<typeparamref name="T" />`

gewarren replied:
Oh yes, sorry about that.

gewarren replied:
I updated the suggestions.

gewarren replied:
> Also, these should have a slash at the end, right? Like `<typeparamref name="T" />`

Yes, good catch.

gewarren commented on xml/System.Runtime.Intrinsics/Vector64`1.xml:
```diff
@@ -37,8 +37,8 @@
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
-    <summary>To be added.</summary>
+    <typeparam name="T">The vector type. <c>T</c> can be any primitive numeric type.</typeparam>
```
```suggestion
    <typeparam name="T">The vector type. <typeparam name="T"> can be any primitive numeric type.</typeparam>
```

dakersnar commented:
@gewarren I added a commit with your suggestions (with an extra backslash that my IDE was complaining about the lack of). I also changed Vector_1.xml in the same way for consistency.

gewarren commented:
> @gewarren I added a commit with your suggestions (with an extra slash that my IDE was complaining about the lack of). I also changed Vector_1.xml in the same way for consistency.

Cool! Which IDE are you using for that functionality?

opbld31 commented:
Learn Build status updates of commit _[36bf5ed](https://github.com/dakersnar/dotnet-api-docs/commits/36bf5ed8c6b8f20a108b93740275fbe7972bfaa7)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.Runtime.Intrinsics/Vector128.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector128.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector128?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector128`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector128%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector128-1?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector256.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector256.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector256?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector256`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector256%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector256-1?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector64.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector64.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector64?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector64`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector64%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector64-1?view=net-7.0&branch=pr-en-us-8464) |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202210052133192858-8464/BuildReport?accessString=3df7c6355f451cecf48755f902b85366fe723c01a85aef7262564c0e8182ff1c).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

dakersnar commented:
VSCode. I'm not an XML expert, but I think it is complaining that `</typeparam>`was closing `<typeparamref name="T">`. It couldn't quite connect the dots to what the real problem was, but adding the slash in `<typeparamref name="T">` fixed it.

![image](https://user-images.githubusercontent.com/18474647/194172214-08e0e143-26e6-463e-9a05-681d46efdebd.png)

opbld32 commented:
Learn Build status updates of commit _[f30eb5f](https://github.com/dakersnar/dotnet-api-docs/commits/f30eb5f04a7c27609e018a44e7cb8151f020215c)_: 

### :x: Validation status: errors

Please follow instructions [here](https://review.learn.microsoft.com/en-us/help/onboard/faq?branch=main#build--publish) which may help to resolve issue.


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[]() | :x:Error | | [Details](#user-content-d89fa3da)
[xml/System.Runtime.Intrinsics/Vector128`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector128%601.xml) | :x:Error | | [Details](#user-content-1e0d6d5afc0bf712eb79d71dca64d4b2-d89fa3da)

<a id="d89fa3da"></a>
###
 - **Line 0, Column 0**: **[Error: ECMA2Yaml_File_LoadFailed]** `````Failed to load 1 files, aborting...`````

<a id="1e0d6d5afc0bf712eb79d71dca64d4b2-d89fa3da"></a>
### [xml/System.Runtime.Intrinsics/Vector128`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector128%601.xml)
 - **Line 0, Column 0**: **[Error: ECMA2Yaml_InternalError]** 
`````
Intenal Several Error: System.Xml.XmlException: Name cannot begin with the '<' character, hexadecimal value 0x3C. Line 40, position 6.
   at System.Xml.XmlTextReaderImpl.ParseQName(Boolean isQName, Int32 startOffset, Int32& colonPos)
   at System.Xml.XmlTextReaderImpl.ParseElement()
   at System.Xml.XmlTextReaderImpl.ParseElementContent()
   at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r)
   at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
   at System.Xml.Linq.XDocument.Parse(String text, LoadOptions options)
   at ECMA2Yaml.ECMALoader.LoadType(FileItem typeFile)
   at ECMA2Yaml.ECMALoader.LoadTypes(String basePath, Namespace ns)
`````

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202210052147551772-8464/BuildReport?accessString=20adda5fd99f8dbd56be15c74f72c547a6dcbe43f675bf6a52a3e43cd17e8c36).

If you see build warnings/errors with permission issues, it might be due to single sign-on (SSO) enabled on Microsoft's GitHub organizations. Please follow instructions [here](https://review.learn.microsoft.com/en-us/help/onboard/sso-troubleshooting?branch=main) to re-authorize your GitHub account to Learn Build.

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

**Note:** Your PR may contain errors or warnings or suggestions unrelated to the files you changed. This happens when external dependencies like GitHub alias, Microsoft alias, cross repo links are updated. Please use these [instructions](https://review.learn.microsoft.com/en-us/help/onboard/faq?branch=main&tabs=pr-validation-error-solution#q-why-do-i-get-pr-validation-result-with-cryptic-error-or-warning-messages-not-related-to-the-changes-i-make-in-my-working-branch) to resolve them.

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

carlossanlop commented on xml/System.Runtime.Intrinsics/Vector128`1.xml:
```diff
@@ -37,8 +37,8 @@
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
-    <summary>To be added.</summary>
+    <<typeparam name="T">>The vector type. <typeparamref name="T" /> can be any primitive numeric type.</typeparam>
```
```suggestion
    <typeparam name="T">The vector type. <typeparamref name="T" /> can be any primitive numeric type.</typeparam>
```

gewarren commented:
@dakersnar I see you were updating the /// in the source code. I found an extension (XML by Red Hat) to do similar validation in the XML file itself. Thanks for the idea!

![image](https://user-images.githubusercontent.com/24882762/194174981-d540a218-299a-4ba4-9f63-1fbb333ce4af.png)

opbld34 commented:
Learn Build status updates of commit _[e6f45b2](https://github.com/dakersnar/dotnet-api-docs/commits/e6f45b2bab42b0fc934e76b62c036b508fc23be5)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.Numerics/Vector`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Numerics/Vector%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.numerics.vector-1?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector128.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector128.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector128?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector128`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector128%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector128-1?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector256.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector256.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector256?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector256`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector256%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector256-1?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector64.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector64.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector64?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector64`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector64%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector64-1?view=net-7.0&branch=pr-en-us-8464) |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202210052211365739-8464/BuildReport?accessString=565c7e16b65829aafb9924884ce826763c76adaf8011db406f62c40eff41ee53).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

tannergooding commented on xml/System.Runtime.Intrinsics/Vector128.xml:
```diff
@@ -18,7 +18,7 @@
   <Interfaces />
   <Docs>
-    <summary>To be added.</summary>
+    <summary>Provides a collection of static convenience methods for creating, manipulating, combining, and converting generic 128-bit vectors.</summary>
```
Same as https://github.com/dotnet/runtime/pull/76687

Combining and converting are a small subset of the total operations supported. Maybe generalizing this as the following would be better:
> Provides a collection of static methods for creating, manipulating, and otherwise operating on 128-bit vectors.

dakersnar commented on xml/System.Numerics/Vector`1.xml:
```diff
@@ -57,7 +57,7 @@
   </Attributes>
   <Docs>
-    <typeparam name="T">The vector type. <typeparamref name="T" /> can be any primitive numeric type.</typeparam>
+    <typeparam name="T">The type of the elements in the vector. <typeparamref name="T" /> can be any primitive numeric type.</typeparam>
```
@tannergooding FYI the `remarks` for this specific case defines what they mean by "primitive numeric type". I removed the term from the other files that don't have that context.

tannergooding replied:
Would probably be good to update the remarks to cover `nint` and `nuint`.

The general table there is also out of date since its really only covering the semantics around `x86/x64` and not other platforms such as `Arm64`.

dakersnar replied:
Would removing the table be the best course of action in that case (see my new commit)? The rest of the remarks section is general enough to apply to both architectures. I suppose it depends on how easy it would be for users to figure out those types with other sources.

carlossanlop replied:
You can remove the remarks table in this PR and resubmit it without urgency on a later date.

Remarks do not show up in intellisense, so they aren't urgent for getting them merged today.

dakersnar replied:
Sounds good. I'll merge this now.

opbld34 commented:
Learn Build status updates of commit _[a0bf75a](https://github.com/dakersnar/dotnet-api-docs/commits/a0bf75a0aefc7c6bef00fe414fad1c2ebf6dc26d)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.Numerics/Vector.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Numerics/Vector.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.numerics.vector?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Numerics/Vector`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Numerics/Vector%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.numerics.vector-1?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector128.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector128.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector128?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector128`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector128%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector128-1?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector256.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector256.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector256?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector256`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector256%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector256-1?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector64.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector64.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector64?view=net-7.0&branch=pr-en-us-8464) |
[xml/System.Runtime.Intrinsics/Vector64`1.xml](https://github.com/dakersnar/dotnet-api-docs/blob/fix-72402/xml/System.Runtime.Intrinsics/Vector64%601.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.vector64-1?view=net-7.0&branch=pr-en-us-8464) |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202210101729337439-8464/BuildReport?accessString=896a4308d5bf341b8000247d520472d9f4a3ed7156ea61072de25fdc9bf884d4).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>
